### PR TITLE
feat(toast): add example of promise toast and add shake animation

### PR DESCRIPTION
**Pull request by juliusmarminge**
Thought I'd add an example of a promise-based toaster.

Adds a shake animation if a toast is updated to destructive:

https://user-images.githubusercontent.com/51714798/232233219-2fc00303-2a31-4546-a4d9-17b790f2dfc4.mp4

### CODE REVIEW

File: apps/www/hooks/use-toast.ts [hunk 1]
```diff
@@ -140,7 +140,7 @@ interface Toast extends Omit<ToasterToast, "id"> {}
 function toast({ ...props }: Toast) {
   const id = genId()
 
-  const update = (props: ToasterToast) =>
+  const update = (props: Toast) =>
```
Comment (juliusmarminge) [inline apps/www/hooks/use-toast.ts:143]: changed the type here as you don't need to pass the `id` when you do x.update()`

**Comment by juliusmarminge** [comment]
cc @shadcn is this something you're interested in?

**Comment by Hidde-Heijnen** [comment]
@juliusmarminge is this something y'all are still working on?

**Comment by juliusmarminge** [comment]
I can pick it up after exams are over. But if you're down to continue now feel free!

**Comment by juliusmarminge** [comment]
not needed now with sonner i guess

**Pull request closed without merging**